lib/tools/node-modules/node-modules-dest-copy.ts

```diff
@@ -46,8 +46,8 @@ export class TnsModulesCopy {
 			shelljs.mkdir("-p", targetDir);
 			shelljs.cp("-Rf", dependency.directory, targetDir);
 
-			//remove platform-specific files (processed separately by plugin services)
-			const targetPackageDir = path.join(targetDir, dependency.name);
+			// remove platform-specific files (processed separately by plugin services)
+			const targetPackageDir = path.join(isScoped ? this.outputRoot : targetDir, dependency.name);
```

Comment:
Now that I look at the method, I'm feeling stupid for not noticing the unnecessary complexity the scoped dep check adds to it. Either I was sleeping while writing that, or shelljs's mkdir didn't work recursively back in October :D, or we were trying to simulate proper scoped dependency dir structure in case the user had npm 2, although I am not sure I am recalling correctly.

The dependency.depth === 0 check was added as an optimization step with the assumption that root-level production node_modules will bring along all other production dependencies, possibly done with npm 2 support in mind.